### PR TITLE
fix: settings layout and behavior fixes

### PR DESCRIPTION
**Pull request by just-toby**
fixes the settings padding and adds a click handler to close the settings modal when the user clicks away

### CODE REVIEW

File: src/components/Swap/Settings/index.tsx [hunk 1]
```diff
@@ -53,6 +58,7 @@ const SettingsButton = styled(IconButton)`
 export default function Settings() {
   const [open, setOpen] = useState(false)
   const [wrapper, setWrapper] = useState<HTMLDivElement | null>(null)
+  useOutsideClickHandler(wrapper, () => setOpen(false))
```
Comment (zzmp) [inline src/components/Swap/Settings/index.tsx:62]: nit: no newline needed

File: src/hooks/useOutsideClickHandler.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export function useOutsideClickHandler(node: HTMLDivElement | null, onOutsideClick: () => void) {
+  const handleClickOutside = (event: MouseEvent) => {
+    if (node && !node.contains(event.target as Node)) {
+      onOutsideClick()
+    }
+  }
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside)
```
Comment (zzmp) [inline src/hooks/useOutsideClickHandler.ts:11]: Do you have concerns that many uses of this hook will cause a lot of global mousedown handlers and eventual performance issues?

File: src/hooks/useOutsideClickHandler.test.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { fireEvent, render, renderHook } from 'test'
+
+import { useOutsideClickHandler } from './useOutsideClickHandler'
+
+describe('useOutsideClickHandler', () => {
+  it('should call onOutsideClick when clicked outside of node', () => {
```
Comment (zzmp) [inline src/hooks/useOutsideClickHandler.test.tsx:6]: Move shared lines to a `beforeEach`.

```suggestion
  let node: HTMLDivElement
  beforeEach(() => {
    node = render(<div />)
    renderHook(() => useOutsideClickHandler(node.container as HTMLDivElement, onOutsideClick))
  })

  it('should call onOutsideClick when clicked outside of node', () => {
```

**Review by zzmp — APPROVED** [review]
This lgtm unless you think the global handlers will be an issue. I think it should be fine.

File: src/hooks/useOutsideClickHandler.ts (hunk 2 above)
Comment (just-toby) [reply src/hooks/useOutsideClickHandler.ts:11]: tbh, i let copilot write this for me 😬   ..... i can optimize it for sure

**Comment by just-toby** [comment]
added jira task to my backlog for optimizing the outside click handler

**Pull request merged**